src/test/emulators/auth/config.spec.ts

```diff
@@ -0,0 +1,161 @@
+      await authApi()
+        .patch(`/identitytoolkit.googleapis.com/v2/projects/${PROJECT_ID}/config`)
+        .set("Authorization", "Bearer owner")
+        .query({ updateMask: "displayName" })
+        .send({})
+        .then((res) => {
+          expectStatusCode(200, res);
+          expect(res.body).not.to.have.property("displayName");
+        });
+    });
+
+    it("performs a full update if the update mask is empty", async () => {
+      await authApi()
+        .patch(`/identitytoolkit.googleapis.com/v2/projects/${PROJECT_ID}/config`)
+        .set("Authorization", "Bearer owner")
+        .send({
+          signIn: { allowDuplicateEmails: true },
+          blockingFunctions: { forwardInboundCredentials: { idToken: true } },
+          usageMode: "PASSTHROUGH",
+        })
+        .then((res) => {
+          expectStatusCode(200, res);
+          expect(res.body.signIn?.allowDuplicateEmails).to.be.true;
+          expect(res.body.blockingFunctions).to.eql({
+            forwardInboundCredentials: { idToken: true },
+          });
+          expect(res.body.usageMode).to.eql("PASSTHROUGH");
+        });
+    });
+
+    it("performs a full update with production defaults if the update mask is empty", async () => {
+      // Update to non-default values
+      await authApi()
+        .patch(`/identitytoolkit.googleapis.com/v2/projects/${PROJECT_ID}/config`)
+        .set("Authorization", "Bearer owner")
+        .send({
+          signIn: { allowDuplicateEmails: true },
+          blockingFunctions: { forwardInboundCredentials: { idToken: true } },
+          usageMode: "PASSTHROUGH",
+        })
+        .then((res) => {
+          expectStatusCode(200, res);
+          expect(res.body.signIn?.allowDuplicateEmails).to.be.true;
+          expect(res.body.blockingFunctions).to.eql({
+            forwardInboundCredentials: { idToken: true },
+          });
+          expect(res.body.usageMode).to.eql("PASSTHROUGH");
+        });
+
+      // Perform a full update and check that production defaults are set
+      await authApi()
+        .patch(`/identitytoolkit.googleapis.com/v2/projects/${PROJECT_ID}/config`)
+        .set("Authorization", "Bearer owner")
+        .send({})
+        .then((res) => {
+          expectStatusCode(200, res);
+          expect(res.body.signIn?.allowDuplicateEmails).to.be.false;
+          expect(res.body.blockingFunctions).to.eql({});
+          expect(res.body.usageMode).to.eql("DEFAULT");
+        });
+    });
+
+    it("should error on unspecified usageMode", async () => {
+      await authApi()
+        .patch(`/identitytoolkit.googleapis.com/v2/projects/${PROJECT_ID}/config`)
+        .set("Authorization", "Bearer owner")
+        .send({
+          usageMode: "USAGE_MODE_UNSPECIFIED",
+        })
+        .then((res) => {
+          expectStatusCode(400, res);
+          expect(res.body.error).to.have.property("message").equals("Invalid usage mode provided");
```

Comment:
Should these error messages be prefixed with an error code? (`SOME_CODE : message`). (You may have to reference production for the right code)